src/Xabe.FFmpeg/Conversion/Conversion.cs

```diff
@@ -281,29 +284,79 @@ public IConversion SetOutput(string outputPath)
+            AddParameter(string.Format("-minrate {0}", bitrate));
+            AddParameter(string.Format("-maxrate {0}", bitrate));
+            AddParameter(string.Format("-bufsize {0}", bitrate));
+
+            if (HasH264Stream())
+                AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
+            // This may be possibly wrong as it assumes if the input stream is X264 and the output is software based x264. 
+            // What about if the output is hardware encoding X264 / X265.
+            // See Method of working out by supplying the output codec type and checking it directly.
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetVideoBitrate(string Minibitrate, string Maxbitrate, string Buffersize, VideoCodec Outputcodec)
+        {
+            AddParameter(string.Format("-b:v {0}", Minibitrate));
+            AddParameter(string.Format("-maxrate {0}", Maxbitrate));
+            AddParameter(string.Format("-bufsize {0}", Buffersize));
+
+            if (Outputcodec == VideoCodec.h264) AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
+
+            return this;
         }
-
-        /// <inheritdoc />
-        public IConversion SetVideoBitrate(long bitrate)
-        {
-            AddParameter(string.Format("-b:v {0}", bitrate));
-            AddParameter(string.Format("-minrate {0}", bitrate));
-            AddParameter(string.Format("-maxrate {0}", bitrate));
-            AddParameter(string.Format("-bufsize {0}", bitrate));
-
-            if (HasH264Stream())
-                AddParameter("-x264opts nal-hrd=cbr:force-cfr=1");
-
-            return this;
-        }
-
         /// <inheritdoc />
         public IConversion SetAudioBitrate(long bitrate)
         {
             AddParameter($"-b:a {bitrate} ");
             return this;
-        }
-
+        } 
+        /// <inheritdoc />
+        public IConversion SetMapChapters(string inputfile)
+        {
+            AddParameter(inputfile == "" ? " -map_chapters -1 " : " -map_chapters " + inputfile);
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetAudioCodecMode(AudioEncoder AE, string bitrate = "", object extraparams = null)
+        {
+            if (AE == AudioEncoder.copy) AddParameter("-c:a copy");
+            if (AE == AudioEncoder.flac) AddParameter($"-c:a {AE}");
+            if ((AE == AudioEncoder.libmp3lame) || (AE == AudioEncoder.lib_aac)) AddParameter($"-c:a aac - q:a {bitrate}");
+            if ((AE == AudioEncoder.libvorbis) || (AE == AudioEncoder.libtwolame) || (AE == AudioEncoder.libwavpack))
+            {
+                AddParameter($"-c:a {AE} " + extraparams.TooParameter());
+            }
+            return this;
+        }
+        /// <inheritdoc />
+        public IConversion SetVideoScaling(string Width, string Height, string Modulas, double aspectratio, Scaling scaler, int x, int y, int left, int top)

```

Comment:
IVideoStream.SetSize modify input stream size.
It should be in VideoStream class but don't know if in this form

Reply:
I have moved it. Please confirm that this will change the output stream not the input stream